### PR TITLE
Update step 34 38 41

### PR DESCRIPTION
**Pull request by drwells**
These are each pretty straightforward.

Part of #6774.

### CODE REVIEW

File: examples/step-34/step-34.cc [hunk 1]
```diff
@@ -1024,9 +1013,8 @@ namespace Step34
       mapping.get_degree(),
       DataOut<dim - 1, DoFHandler<dim - 1, dim>>::curved_inner_cells);
 
-    std::string filename =
-      (Utilities::int_to_string(dim) + "d_boundary_solution_" +
-       Utilities::int_to_string(cycle) + ".vtk");
+    std::string filename = std::to_string(dim) + "d_boundary_solution_" +
```
Comment (jppelteret) [inline examples/step-34/step-34.cc]: Nitpick: Couldn't this be made `const`?

File: examples/step-38/doc/intro.dox [hunk 2]
```diff
@@ -118,15 +118,15 @@ This surface description given, in the codimension one surface case, the two
 routines FEValues::shape_grad and FEValues::JxW
 return
 @f{align*}
-\text{FEValues::shape\_grad}(i,l)&=D \mathbf x_K(p_l) G^{-1}(p_l)D(\varphi_i \circ \mathbf x_K)
+\text{FEValues::shape_grad}(i,l)&=D \mathbf x_K(p_l) G^{-1}(p_l)D(\varphi_i \circ \mathbf x_K)
```
Comment (jppelteret) [inline examples/step-38/doc/intro.dox]: Nice catch!

File: examples/step-34/step-34.cc (hunk 1 above)
Comment (drwells) [reply examples/step-34/step-34.cc]: yup. Lets do it!

File: examples/step-38/doc/intro.dox (hunk 2 above)
Comment (drwells) [reply examples/step-38/doc/intro.dox]: Its odd that mathjax renders the backslash here.

**Pull request merged**